controller/workitem.go

```diff
@@ -111,18 +139,45 @@ func (c *WorkitemController) Update(ctx *app.UpdateWorkitemContext) error {
 	if !authorized {
 		return jsonapi.JSONErrorResponse(ctx, errors.NewForbiddenError("user is not authorized to access the space"))
 	}
+
+	if ctx.Payload.Data.Relationships != nil && ctx.Payload.Data.Relationships.BaseType != nil &&
+		ctx.Payload.Data.Relationships.BaseType.Data != nil && ctx.Payload.Data.Relationships.BaseType.Data.ID != wi.Type {
+
+		authorized, err := c.authorizeWorkitemTypeEditor(ctx, c.db, wi.SpaceID, creator.(string), currentUserIdentityID.String())
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+		if !authorized {
+			return jsonapi.JSONErrorResponse(ctx, errors.NewForbiddenError("user is not authorized to change the workitemtype"))
+		}
```

Comment:
Shouldn't there be a check that the new workitemtype should be an allowed workitemtype e.g. belongs to the same space template.

Reply:
We check this here https://github.com/fabric8-services/fabric8-wit/pull/2202/files#diff-55f2da878ed2fcb42e5373010f02ff92R1168